xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-default/src/main/java/org/xwiki/platform/security/requiredrights/internal/analyzer/DefaultObjectRequiredRightAnalyzer.java

```diff
@@ -0,0 +1,208 @@
+            }
+
+            return results;
+        }
+    }
+
+    protected List<RequiredRightAnalysisResult> analyzeProperty(BaseObject object, String propertyName,
+        BaseClass xClass) throws RequiredRightsException
+    {
+        PropertyInterface property = xClass.getField(propertyName);
+        if (property instanceof TextAreaClass) {
+            TextAreaClass textAreaClass = (TextAreaClass) property;
+
+            return analyzeTextAreaProperty(object, propertyName, textAreaClass);
+        }
+        return List.of();
+    }
+
+    protected List<RequiredRightAnalysisResult> analyzeTextAreaProperty(BaseObject object, String propertyName,
+        TextAreaClass textAreaClass) throws RequiredRightsException
+    {
+        String contentTypeString = textAreaClass.getContentType();
+        TextAreaClass.ContentType contentType =
+            TextAreaClass.ContentType.getByValue(contentTypeString);
+        PropertyInterface field = object.getField(propertyName);
+
+        List<RequiredRightAnalysisResult> result = List.of();
+
+        // No need to analyze restricted properties.
+        if (!textAreaClass.isRestricted() && field instanceof BaseStringProperty) {
+            String value = ((BaseStringProperty) field).getValue();
+
+            if (contentType != null) {
+                switch (contentType) {
+                    case VELOCITY_CODE:
+                        result = analyzeVelocityScriptValue(value, field.getReference(),
+                            "security.requiredrights.object.velocityCodeTextArea");
+                        break;
+                    case VELOCITYWIKI:
+                        result = analyzeVelocityScriptValue(value, field.getReference(),
+                            "security.requiredrights.object.velocityWikiTextArea");
+                        if (result.isEmpty()) {
+                            // If there is no Velocity code, we analyze the content as wiki syntax.
+                            result = analyzeWikiContent(object, value, field.getReference());
+                        }
+                        break;
+                    case WIKI_TEXT:
+                        result = analyzeWikiContent(object, value, field.getReference());
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    private List<RequiredRightAnalysisResult> analyzeVelocityScriptValue(String value, EntityReference reference,
+        String translationMessage)
+    {
+        List<RequiredRightAnalysisResult> result;
+        if (VelocityUtil.containsVelocityScript(value)) {
+            result = List.of(new RequiredRightAnalysisResult(reference,
+                this.translationMessageSupplierProvider.get(translationMessage),
+                this.stringCodeBlockSupplierProvider.get(value),
+                List.of(
```

Comment:
This List should probably be a constant to avoid recreating it all the time.

Reply:
I've added constants to `RequiredRight` for common cases.